From 4351da740efb899d067abee8a1fdb2aacaf9f5e6 Mon Sep 17 00:00:00 2001
From: "Ethan D. Twardy" <ethan.twardy@gmail.com>
Date: Wed, 11 Oct 2023 21:21:36 -0500
Subject: [PATCH 3/3] wayland: Setup sway desktop for work virtual machines

 * Disable hardware cursors
 * Don't configure the apple-dcp driver
 * Use agreety instead of tuigreet
 * Remove 50-systemd-user.conf (already installed on Arch)
 * Remove ssh-agent.service (already installed on Arch)
 * Switch greetd.toml to use "greeter" user instead of "greetd" user,
   since that's the name of the user created by the Arch package
   maintainers.
 * Use sway/workspaces waybar module
 * Use the default sway background
 * Explicitly set the output resolution to 1920x1080
---
 Makefile                     |  6 ------
 home/.config/sway/config     |  5 ++++-
 home/.config/waybar/config   |  1 +
 ssh-agent.service            | 10 ----------
 wayland/50-systemd-user.conf |  9 ---------
 wayland/apple-dcp.conf       |  1 -
 wayland/cobblede.sh          |  1 +
 wayland/greetd.toml          |  4 ++--
 8 files changed, 8 insertions(+), 29 deletions(-)
 delete mode 100644 ssh-agent.service
 delete mode 100644 wayland/50-systemd-user.conf
 delete mode 100644 wayland/apple-dcp.conf

diff --git a/Makefile b/Makefile
index c30bd2d..5c28c47 100644
--- a/Makefile
+++ b/Makefile
@@ -19,9 +19,3 @@ install:
 		-t $(DESTDIR)$(systemd_userdir)/xdg-desktop-portal-gtk.service.d
 	install -Dm644 wayland/xdg-desktop-portal-wlr.conf \
 		-t $(DESTDIR)$(systemd_userdir)/xdg-desktop-portal-wlr.service.d
-	: # Ensure critical environment variables transfer from sway
-	install -Dm644 wayland/50-systemd-user.conf -t $(DESTDIR)/etc/sway/config.d
-	: # SSH Agent
-	install -Dm644 ssh-agent.service -t $(DESTDIR)$(systemd_userdir)
-	: # Enable "the notch"
-	install -Dm644 wayland/apple-dcp.conf -t $(DESTDIR)/lib/modprobe.d
diff --git a/home/.config/sway/config b/home/.config/sway/config
index cbddf5a..9cb89f1 100644
--- a/home/.config/sway/config
+++ b/home/.config/sway/config
@@ -31,7 +31,10 @@ set $lock gtklock -s ~/.config/gtklock/style.css
 # Output
 ###
 
-output * bg /usr/share/backgrounds/sway-background.png fill scale 2
+output * {
+  bg /usr/share/backgrounds/sway/Sway_Wallpaper_Blue_1920x1080.png fill
+  resolution 1920x1080
+}
 
 ###############################################################################
 # Idle
diff --git a/home/.config/waybar/config b/home/.config/waybar/config
index c22c57d..beea750 100644
--- a/home/.config/waybar/config
+++ b/home/.config/waybar/config
@@ -2,6 +2,7 @@
     "layer": "top",
     "height": 34,
     "position": "top",
+    "modules-left": ["sway/workspaces"],
     "modules-right": ["wireplumber", "idle_inhibitor", "network", "battery", "clock"],
     "clock": {
         "format": "{:%d %b %H:%M}",
diff --git a/ssh-agent.service b/ssh-agent.service
deleted file mode 100644
index 9428410..0000000
--- a/ssh-agent.service
+++ /dev/null
@@ -1,10 +0,0 @@
-[Unit]
-Description=SSH key agent
-
-[Service]
-Type=simple
-Environment=SSH_AUTH_SOCK=%t/ssh-agent.socket
-ExecStart=/usr/bin/ssh-agent -D -a $SSH_AUTH_SOCK
-
-[Install]
-WantedBy=default.target
diff --git a/wayland/50-systemd-user.conf b/wayland/50-systemd-user.conf
deleted file mode 100644
index ebbf37b..0000000
--- a/wayland/50-systemd-user.conf
+++ /dev/null
@@ -1,9 +0,0 @@
-# sway does not set DISPLAY/WAYLAND_DISPLAY in the systemd user environment
-# See FS#63021
-# Adapted from xorg's 50-systemd-user.sh, which achieves a similar goal.
-
-exec systemctl --user import-environment \
-     DISPLAY WAYLAND_DISPLAY SWAYSOCK XDG_CURRENT_DESKTOP
-exec hash dbus-update-activation-environment 2>/dev/null \
-     && dbus-update-activation-environment --systemd \
-     DISPLAY WAYLAND_DISPLAY SWAYSOCK XDG_CURRENT_DESKTOP
diff --git a/wayland/apple-dcp.conf b/wayland/apple-dcp.conf
deleted file mode 100644
index 74e4e9b..0000000
--- a/wayland/apple-dcp.conf
+++ /dev/null
@@ -1 +0,0 @@
-options apple_dcp show_notch=Y
diff --git a/wayland/cobblede.sh b/wayland/cobblede.sh
index f8a01bb..482f96e 100644
--- a/wayland/cobblede.sh
+++ b/wayland/cobblede.sh
@@ -5,3 +5,4 @@ export SDL_VIDEODRIVER=wayland
 export _JAVA_AWT_WM_NONREPARENTING=1
 export XDG_SESSION_DESKTOP=sway
 export XDG_CURRENT_DESKTOP=sway
+export WLR_NO_HARDWARE_CURSORS=1
diff --git a/wayland/greetd.toml b/wayland/greetd.toml
index 9fc7a79..1e574c8 100644
--- a/wayland/greetd.toml
+++ b/wayland/greetd.toml
@@ -8,9 +8,9 @@ vt = 1
 
 # `agreety` is the bundled agetty/login-lookalike. You can replace `/bin/sh`
 # with whatever you want started, such as `sway`.
-command = "tuigreet --cmd 'systemd-cat --identifier=wayland /usr/bin/sway'"
+command = "agreety --cmd 'systemd-cat --identifier=wayland /usr/bin/sway'"
 
 # The user to run the command as. The privileges this user must have depends
 # on the greeter. A graphical greeter may for example require the user to be
 # in the `video` group.
-user = "greetd"
+user = "greeter"
-- 
2.44.0

